addons/mass_mailing/static/src/builder/options/snippet_visibility_option.js

```diff
@@ -0,0 +1,81 @@
+import { BaseOptionComponent, useDomState } from "@html_builder/core/utils";
+import { Domain } from "@web/core/domain";
+import { DomainSelectorDialog } from "@web/core/domain_selector_dialog/domain_selector_dialog";
+import { _t } from "@web/core/l10n/translation";
+import { useService } from "@web/core/utils/hooks";
+
+/**
+ * Option Component to enable the user to filter some snippets to a specific subset of ids following
+ * a domain.
+ */
+export class SnippetVisibilityOption extends BaseOptionComponent {
+    static template = "mass_mailing.VisibilityOption";
+    static props = { getModel: Function };
+
+    setup() {
+        super.setup();
+        this.treeProcessor = useService("tree_processor");
+        this.dialog = useService("dialog");
+        this.historyPlugin = this.env.editor.shared.history;
+        this.overlayButtonsPlugin = this.env.editor.shared.overlayButtons;
+
+        this.state = useDomState((editingElement) => {
+            const currentDomain = new Domain(
+                JSON.parse(editingElement.dataset.filterDomain || "[]")
+            );
+            this.parseTree(currentDomain);
+            if (currentDomain.toJson().length) {
+                editingElement.children[0]?.setAttribute(
+                    "data-tooltip",
+                    _t("Visible Condtionnally")
+                );
+            } else {
+                editingElement.children[0]?.removeAttribute("data-tooltip");
+            }
+            return {
+                domain: currentDomain,
+            };
+        });
+    }
+
+    get stringifiedDomain() {
+        return JSON.stringify(this.state.domain.toJson());
+    }
+
+    /**
+     *
+     * @param {import("@web/core/tree_editor/condition_tree").Tree} tree
+     */
+    async parseTree(domain) {
+        const resModel = this.props.getModel();
+        const tree = await this.treeProcessor.treeFromDomain(resModel, domain, !this.env.debug);
+        const trees = !tree.negate && tree.value === "&" ? tree.children : [tree];
```

Comment:
I can see this line in `domain_field.js`, and a variant in `search_model.js`
```js
        const trees =
            !tree.negate &&
            tree.type === "connector" &&
            tree.value === "&" &&
            tree.children.length > 0
                ? tree.children
                : [tree];
```
I suppose this is because Odoo facets represent leafs separated by an `&`, so we want to get all first-depth leaves (tree.children) if the tree value is `&`, but if it is a `OR`, we consider that this is only one facet? And I guess in our case, it is not really possible to have a tree negated at the root?